dropwizard-jersey/src/main/java/io/dropwizard/jersey/optional/OptionalLongMessageBodyWriter.java

```diff
@@ -36,7 +37,7 @@ public void writeTo(OptionalLong entity,
                         MultivaluedMap<String, Object> httpHeaders,
                         OutputStream entityStream) throws IOException {
         if (!entity.isPresent()) {
-            throw new NotFoundException();
+            throw new NoContentException("Empty content.");
```

Comment:
Actually, [looking at the code](https://github.com/dropwizard/dropwizard/blob/a0c5e2ab97a91200e09c91397fab0a0c359b8483/dropwizard-jersey/src/main/java/io/dropwizard/jersey/errors/LoggingExceptionMapper.java#L22-L32), I believe even if you supplied a `WebApplicationException` with a fully built response, it will respond with a _json payload_ and a no content status 😢 

I'm not sure if this is the intended purpose of `WebApplicationException`. Let me ponder this.